### PR TITLE
Fix for homebrew and support for non-standard python distributions

### PR DESCRIPTION
**Pull request by MisterTea**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -55,7 +55,7 @@ BLASLDFLAGSSO ?= $(BLASLDFLAGS)
 # pure C++ test in the test directory
 
 tests/test_blas: tests/test_blas.cpp
-	$(CC) $(CFLAGS) $< -o $@ $(BLASLDFLAGS) $(BLASCFLAGS)
+	$(CC) $(CFLAGS) $< -o $@ $(LDFLAGS) $(BLASLDFLAGS) $(BLASCFLAGS)
```
Comment (mdouze) [inline Makefile]: Good catch

File: example_makefiles/makefile.inc.Mac.brew [hunk 2]
```diff
@@ -5,7 +5,7 @@
 # brew install llvm
 CC=/usr/local/opt/llvm/bin/clang++
 CFLAGS=-fPIC -m64 -Wall -g -O3 -msse4 -mpopcnt -fopenmp -Wno-sign-compare -I/usr/local/opt/llvm/include -std=c++11
-LDFLAGS=-g -fPIC -fopenmp -L/usr/local/opt/llvm/lib
+LDFLAGS=-g -fPIC -fopenmp -L/usr/local/opt/llvm/lib -L/usr/local/Cellar/llvm/5.0.0/lib
```
Comment (mdouze) [inline example_makefiles/makefile.inc.Mac.brew]: Why is this needed?

File: example_makefiles/makefile.inc.Mac.brew [hunk 3]
```diff
@@ -77,17 +77,11 @@ BLASLDFLAGS=-framework Accelerate
...
-# python -c "import numpy ; print numpy.get_include()"
-#
-# the paths below are for the system python (not the macports one)
-
-PYTHONCFLAGS=-I/System/Library/Frameworks/Python.framework/Versions/2.7/include/python2.7 \
--I/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/numpy/core/include
+# The Python include directories for the current python executable
 
+PYTHON_INC = $(shell python -c "from __future__ import print_function; import distutils.sysconfig; print(distutils.sysconfig.get_python_inc())")
+NUMPY_INC = $(shell python -c "from __future__ import print_function; import numpy ; print(numpy.get_include())")
```
Comment (mdouze) [inline example_makefiles/makefile.inc.Mac.brew]: you don't need the from __future__ because the print has just a single argument, ie. `print (x)` woks fine in py2 as well.

File: example_makefiles/makefile.inc.Mac.brew (hunk 2 above)
Comment (MisterTea) [reply example_makefiles/makefile.inc.Mac.brew]: Because homebrew doesn't install libomp to /usr/local/lib

**Comment by MisterTea** [comment]
Feedback addressed.  Also changed it to handle any version of LLVM.
